WhisperAPI/WhisperAPI/Controllers/ContextController.cs

```diff
@@ -21,20 +21,29 @@ public ContextController(IContexts contexts)
 
         public override void OnActionExecuting(ActionExecutingContext actionExecutingContext)
         {
-            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
+            Guid chatKey;
+            object param;
+            if (actionExecutingContext.ActionArguments.TryGetValue("searchQuery", out param))
```

Comment:
The code structure is getting a bit messy. We should have something like:

public override void onActionExecuting...
{
    // Model validation
    Guid ChatKey;
    if(try to get the search query)
        ChatKey = searchquery.ChatKey;
    else if (try to get ChatKey directly from action arguments)
        ChatKey = ChatKey;
    else
        // We couldn't get the ChatKey... Log an error
    // Rest of the method (retrieving context)
}